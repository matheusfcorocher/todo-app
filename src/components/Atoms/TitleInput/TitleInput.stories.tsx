import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import TitleInput from './TitleInput';
import { Default as DeleteIcon } from '../icons/DeleteIcon/DeleteIcon.stories';

export default {
  title: 'Atoms/TitleInput',
  component: TitleInput,
  decorators: [
    (Story) => (
      <ul style={{ listStyleType: 'none', margin: 0, padding: 0 }}>
        <Story />
      </ul>
    ),
  ]
} as ComponentMeta<typeof TitleInput>;

const Template: ComponentStory<typeof TitleInput> = (args) => 
  <TitleInput props={{
    handleOnChange: function (): void {
      throw new Error('Function not implemented.');
    },
    handleOnFocus: function (): void {
      throw new Error('Function not implemented.');
    },
    handleOnBlur: function (): void {
      throw new Error('Function not implemented.');
    },
    title: '',
    className: undefined
  }} />;

export const Default = Template.bind({});
Default.args = {};