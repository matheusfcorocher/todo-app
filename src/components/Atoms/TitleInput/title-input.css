.title {
    width: 100%;
    margin: 0 0.5rem 0 0.5rem;
}

.EDITING .title {
    width: 90%;
}

.title-input {
    display: flex;
    align-items: center;
    height: 100%;
    width: 100%;
    padding: 0.5rem;
    border: none;
    text-overflow: ellipsis;
    font-family: 'Nunito Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif;
    font-size: 1rem;
    /* font-weight: 400; */
}

.true .title-input {
    text-decoration: line-through;
    color: green;
}
