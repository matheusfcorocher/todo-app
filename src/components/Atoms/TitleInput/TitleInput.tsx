import React, { useEffect, useRef, useState } from 'react';
import '../../../App';
import './title-input.css';

interface TitleInputProps {
    handleOnChange: () => void,
    handleOnFocus: () => void,
    handleOnBlur: () => void,
    title:  string,
    ref?:  React.MutableRefObject<HTMLInputElement>,
    className?: string,
}

function TitleInput({
    handleOnChange,
    handleOnFocus,
    handleOnBlur,
    title,
    ref,
    className
}: TitleInputProps) {
    return (
        <label
            htmlFor="title"
            aria-label={title}
            className="title"
        >
            <input
                type="text"
                role="textbox"
                value={title}
                className={`title-input`}
                ref={ref}
                onChange={() => handleOnChange()}
                onFocus={() => handleOnFocus()}
                onBlur={() => handleOnBlur()}
            />
        </label>
    );
}

export default TitleInput;
