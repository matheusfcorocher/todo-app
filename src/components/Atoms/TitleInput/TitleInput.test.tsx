import React, { useState } from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import TitleInput from './TitleInput';

describe("::Components ::Atoms ::TitleInput", () => {
    test('renders TitleInput', () => {
        render(<TitleInput handleOnChange={function (): void {
            throw new Error('Function not implemented.');
        }}
            handleOnFocus={function (): void {
                throw new Error('Function not implemented.');
            }}
            handleOnBlur={function (): void {
                throw new Error('Function not implemented.');
            }}
            title={''}
            className={undefined} />);
        const titleInput = screen.getByRole("textbox");
        expect(titleInput).toMatchSnapshot();
    })
})
