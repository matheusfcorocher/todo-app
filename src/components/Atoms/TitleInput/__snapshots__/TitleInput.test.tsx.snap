// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`::Components ::Atoms ::TitleInput renders TitleInput 1`] = `
<input
  class="title-input"
  role="textbox"
  type="text"
  value=""
/>
`;
